src/runtime_src/hip/core/memory.cpp

```diff
@@ -57,11 +57,18 @@ namespace xrt::core::hip
       // TODO Need to create locked memory for Default and Portable flags. Creating a regular BO for now
       case hipHostMallocDefault:
       case hipHostMallocPortable:
-      case hipHostMallocMapped:
-      case hipHostMallocWriteCombined:
+      case hipHostMallocMapped: {
         init_xrt_bo();
         break;
-
+      }
+      case hipHostMallocWriteCombined: {
+        // This is a workaround to create a buffer with cacheable flag if WriteComined flag is provided. 
+        // This gets used to create instruction buffer on NPU
+        // TODO This would go away once xrt::elf flow is enabled
+        auto xrt_device = m_device->get_xrt_device();
+        m_bo = std::make_shared<xrt::bo>(xrt_device, m_size, xrt::bo::flags::cacheable, 1);
```

Comment:
I am not following the need to create xrt::bo as a shared_ptr?  xrt::bo is a reference counted impl already.

Reply:
Yes Soren. We dont require it, I thought of changing in my last PR itself, let me change it in this PR.

Reply:
Fixed the issue. Fixed function signatures of some of hip memory class aswell due to this change.
Tried making hip::memory::read as const, but it is calling xrt::bo::read which is non-const, thatswhy i have to remove the const-ness of this function.